pi5/pinctrl_utils.go

```
@@ -0,0 +1,231 @@
+//go:build linux
+
+package pi5
+
+import (
+	"encoding/binary"
+	"fmt"
+	"math"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+type rangeInfo struct {
+	childAddr     uint64
+	parentAddr    uint64
+	addrSpaceSize uint64
+}
+
+type gpioChip struct {
+	name     string
+	dtNode   string
+	physAddr uint64
+	virtAddr uint64
+	chipSize uint32
+	// memfd     uint32
+}
+
+var INVALID_ADDR uint64 = math.MaxUint64
+
+const gpioName = "gpio0"
+const dtBaseNodePath = "/proc/device-tree"
+
+// Sets up GPIO Pin Memory Access by parsing the device tree for relevant address information
+func (b *pinctrlpi5) pinControlSetup() error {
+	nodePath, err := b.findPathFromAlias(gpioName) // this ("gpio") is hardcoded now, we will fix that later!
+	if err != nil {
+		b.logger.Errorf("error getting raspi5 GPIO nodePath")
+	}
+
+	err = b.setGPIONodePhysAddr(nodePath)
+	if err != nil {
+		b.logger.Errorf("error getting raspi5 GPIO physical address")
```

same here we should return the error